robusta_krr/core/integrations/kubernetes.py

```diff
@@ -216,9 +238,20 @@ async def __list_hpa(self) -> dict[tuple[str, str], HPAData]:
         """
 
         loop = asyncio.get_running_loop()
-        res: V2HorizontalPodAutoscalerList = await loop.run_in_executor(
-            self.executor, lambda: self.autoscaling.list_horizontal_pod_autoscaler_for_all_namespaces(watch=False)
-        )
+
+        try:
+            # Try to use V2 API first
+            res: V2HorizontalPodAutoscalerList = await loop.run_in_executor(
```

Comment:
do we need to return the result?

Reply:
do we have to format the result to be of object HPAData

Reply:
It is formatted below. 
`__get_metric` is the inner function